spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarProperties.java

```diff
@@ -375,63 +374,65 @@ public void setRegexSubscriptionMode(RegexSubscriptionMode regexSubscriptionMode
 		}
 
-		public boolean isReplicateSubscriptionState() {
+		public Boolean isReplicateSubscriptionState() {
 			return this.replicateSubscriptionState;
 		}
 
-		public void setReplicateSubscriptionState(boolean replicateSubscriptionState) {
+		public void setReplicateSubscriptionState(Boolean replicateSubscriptionState) {
 			this.replicateSubscriptionState = replicateSubscriptionState;
 		}
 
-		public boolean isAutoAckOldestChunkedMessageOnQueueFull() {
+		public Boolean isAutoAckOldestChunkedMessageOnQueueFull() {
 			return this.autoAckOldestChunkedMessageOnQueueFull;
 		}
 
-		public void setAutoAckOldestChunkedMessageOnQueueFull(boolean autoAckOldestChunkedMessageOnQueueFull) {
+		public void setAutoAckOldestChunkedMessageOnQueueFull(Boolean autoAckOldestChunkedMessageOnQueueFull) {
 			this.autoAckOldestChunkedMessageOnQueueFull = autoAckOldestChunkedMessageOnQueueFull;
 		}
 
-		public int getMaxPendingChunkedMessage() {
+		public Integer getMaxPendingChunkedMessage() {
 			return this.maxPendingChunkedMessage;
 		}
 
-		public void setMaxPendingChunkedMessage(int maxPendingChunkedMessage) {
+		public void setMaxPendingChunkedMessage(Integer maxPendingChunkedMessage) {
 			this.maxPendingChunkedMessage = maxPendingChunkedMessage;
 		}
 
-		public long getExpireTimeOfIncompleteChunkedMessageMillis() {
-			return this.expireTimeOfIncompleteChunkedMessageMillis;
+		public Duration getExpireTimeOfIncompleteChunkedMessage() {
+			return this.expireTimeOfIncompleteChunkedMessage;
 		}
 
-		public void setExpireTimeOfIncompleteChunkedMessageMillis(long expireTimeOfIncompleteChunkedMessageMillis) {
-			this.expireTimeOfIncompleteChunkedMessageMillis = expireTimeOfIncompleteChunkedMessageMillis;
+		public void setExpireTimeOfIncompleteChunkedMessage(Duration expireTimeOfIncompleteChunkedMessage) {
+			this.expireTimeOfIncompleteChunkedMessage = expireTimeOfIncompleteChunkedMessage;
 		}
 
 		public Map<String, Object> buildProperties() {
 			PulsarProperties.Properties properties = new Properties();
 
 			PropertyMapper map = PropertyMapper.get().alwaysApplyingWhenNonNull();
 
-			map.from(this::getTopics).as(Set::of).to(properties.in("topicNames"));
-			map.from(this::getTopicsPattern).as(Pattern::compile).to(properties.in("topicsPattern"));
+			map.from(this::getTopics).to(properties.in("topicNames"));
+			map.from(this::getTopicsPattern).to(properties.in("topicsPattern"));
 			map.from(this::getSubscriptionName).to(properties.in("subscriptionName"));
 			map.from(this::getSubscriptionType).to(properties.in("subscriptionType"));
 			map.from(this::getReceiverQueueSize).to(properties.in("receiverQueueSize"));
-			map.from(this::getAcknowledgementsGroupTimeMicros).to(properties.in("acknowledgementsGroupTimeMicros"));
-			map.from(this::getNegativeAckRedeliveryDelayMicros).to(properties.in("negativeAckRedeliveryDelayMicros"));
+			map.from(this::getAcknowledgementsGroupTime).as(it -> it.toNanos() / 1000)
```

Comment:
This (the PR as a whole) is a change that would benefit from #176 being in place beforehand. Not saying to go do that for this PR - just saying out loud. 

I don't think these changes are too risky though and we also have fair test coverage (indirectly). I do remember some issue w/ the mapper not doing what I expected it to do when using `as` (even though we said "only apply when non-null" it was still executing). May be worth checking this particular case for a property using `as`.

Reply:
Yes I think adding tests is the next step.